File: data-structures/arrays/python/array_2D_update_constants.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


# Index values of each level
LEVEL1 = 0 
LEVEL2 = 1
LEVEL3 = 2


def main():
    # Declare and initialise the two-dimensional list of words
    spelling_words = [
        ["me", "do", "her", "it", "him"],
        ["put", "ask", "says", "red", "any"],
        ["they", "where", "friend", "fast", "class"]
    ]

    # Ask the user for the new word
    new_word = input("Enter the new word: ")

    # Store the original word for Level 3 at index 4
    word_index = 4
    old_word = spelling_words[LEVEL3][word_index]

    # Update the word for Level 3 at index 4
    spelling_words[LEVEL3][word_index] = new_word

    # Output the updated word
    updated_word = spelling_words[LEVEL3][word_index]
    print(f"The word {old_word} has been updated to {updated_word}")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()

File: data-structures/arrays/java/array_2D_update_constants.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Scanner;

class Arrays2D  {
    // Index values of each level
    public static final int LEVEL_1 = 0;
    public static final int LEVEL_2 = 1;
    public static final int LEVEL_3 = 2;

    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        // Declare and initialise the two-dimensional array of words
        String[][] spellingWords = {
            { "me", "do", "her", "it", "him" },
            { "put", "ask", "says", "red", "any" },
            { "they", "where", "friend", "fast", "class" }
        };

        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        // Ask the user for the new word
        System.out.println("Enter the new word:");
        String newWord = scanner.nextLine();

        // Store the original word for Level 3 at index 4
        int wordIndex = 4;
        String oldWord = spellingWords[LEVEL_3][wordIndex];

        // Update the word for Level 3 at index 4
        spellingWords[LEVEL_3][wordIndex] = newWord;

        // Output the updated word
        String updatedWord = spellingWords[LEVEL_3][wordIndex];
        System.out.println("The word " + oldWord + " has been updated to " + updatedWord);
    }
}